src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/BrowserWebSocket.cs

```diff
@@ -0,0 +1,529 @@
+                }
+            }
+        }
+
+        private void ConnectExceptionCleanup()
+        {
+            Dispose();
+        }
+
+        public override void Dispose()
+        {
+            int priorState = Interlocked.Exchange(ref _state, _disposed);
+            if (priorState == _disposed)
+            {
+                // No cleanup required.
+                return;
+            }
+
+            // registered by the CancellationTokenSource cts in the connect method
+            _cts.Cancel(false);
+            _cts.Dispose();
+
+            _writeBuffer?.Dispose();
+
+            // We need to clear the events on websocket as well or stray events
+            // are possible leading to crashes.
+            if (_onClose != null)
+            {
+                _innerWebSocket?.SetObjectProperty("onclose", "");
+                JavaScript.Runtime.FreeObject(_onClose);
+            }
+            if (_onError != null)
+            {
+                _innerWebSocket?.SetObjectProperty("onerror", "");
+                JavaScript.Runtime.FreeObject(_onError);
+            }
+            if (_onOpen != null)
+            {
+                _innerWebSocket?.SetObjectProperty("onopen", "");
+                JavaScript.Runtime.FreeObject(_onOpen);
+            }
+            if (_onMessage != null)
+            {
+                _innerWebSocket?.SetObjectProperty("onmessage", "");
+                JavaScript.Runtime.FreeObject(_onMessage);
+            }
+            _innerWebSocket?.Dispose();
+        }
+
+        // This method is registered by the CancellationTokenSource cts in the connect method
+        // and called by Dispose or Abort so that any open websocket connection can be closed.
+        private void AbortRequest()
+        {
+            if (State == WebSocketState.Open)
+            {
+                var closeResult = CloseAsyncCore(WebSocketCloseStatus.NormalClosure, "Connection was aborted", CancellationToken.None);
+            }
+        }
+
+        /// <summary>
+        /// Send data on <see cref="System.Net.WebSockets.ClientWebSocket"/> as an asynchronous operation.
+        /// </summary>
+        /// <returns>The async.</returns>
+        /// <param name="buffer">Buffer.</param>
+        /// <param name="messageType">Message type.</param>
+        /// <param name="endOfMessage">If set to <c>true</c> end of message.</param>
+        /// <param name="cancellationToken">Cancellation token.</param>
+        public override async Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType messageType, bool endOfMessage, CancellationToken cancellationToken)
+        {
+            ThrowIfNotConnected();
+
+            if (messageType != WebSocketMessageType.Binary &&
+                messageType != WebSocketMessageType.Text)
+            {
+                throw new ArgumentException($"Invalid message type: '{messageType}' specified in method 'SendAsync'.  Valid types are 'Binary' and 'Text'",
```

Comment:
Exception messages should be in the resx

Reply:
Addressed